heat/core/arithmetics.py

```diff
@@ -129,6 +132,72 @@ def _add(self, other):
 DNDarray.__radd__.__doc__ = add.__doc__
 
 
+def add_(t1: DNDarray, t2: Union[DNDarray, float]) -> DNDarray:
+    """
+    Element-wise in-place addition of values of two operands.
+    Takes the first operand (:class:`~heat.core.dndarray.DNDarray`) and element-wise adds the
+    element(s) of the second operand (scalar or :class:`~heat.core.dndarray.DNDarray`) in-place,
+    i.e. the element(s) of `t1` are overwritten by the results of element-wise addition of `t1` and
+    `t2`.
+    Can be called as a DNDarray method or with the symbol `+=`.
+
+    Parameters
+    ----------
+    t1: DNDarray
+        The first operand involved in the addition
+    t2: DNDarray or scalar
+        The second operand involved in the addition
+
+    Raises
+    ------
+    ValueError
+        If both inputs are DNDarrays that do not have the same split axis and the shapes of their
+        underlying torch.tensors differ, s.t. we cannot process them directly without resplitting.
+    TypeError
+        If the data type of `t2` cannot be cast to the data type of `t1`. Although the
+        corresponding out-of-place operation may work, for the in-place version the requirements
+        are stricter, because the data type of `t1` does not change.
+
+    Examples
+    --------
+    >>> import heat as ht
+    >>> T1 = ht.float32([[1, 2], [3, 4]])
+    >>> T2 = ht.float32([[2, 2], [2, 2]])
+    >>> T1 += T2
+    >>> T1
+    DNDarray([[3., 4.],
+              [5., 6.]], dtype=ht.float32, device=cpu:0, split=None)
+    >>> T2
+    DNDarray([[2., 2.],
+              [2., 2.]], dtype=ht.float32, device=cpu:0, split=None)
+    >>> s = 2.0
+    >>> T2.add_(s)
+    DNDarray([[4., 4.],
+              [4., 4.]], dtype=ht.float32, device=cpu:0, split=None)
+    >>> T2
+    DNDarray([[4., 4.],
+              [4., 4.]], dtype=ht.float32, device=cpu:0, split=None)
+    >>> s
+    2.0
+    """
+
+    def wrap_add_(a: torch.Tensor, b: torch.Tensor) -> torch.Tensor:
+        return a.add_(b)
+
+    try:
+        return _operations.__binary_op(wrap_add_, t1, t2, out=t1)
+    except NotImplementedError:
+        raise ValueError(
+            f"The differently split inputs (splits {t1.split} and {t2.split} cannot be processed "
+            + "in-place without resplitting, because the underlying tensors have shapes which are "
+            + f"not broadcastable (shapes {t1.larray.shape} and {t2.larray.shape})."
+        )
```

Comment:
```suggestion
            f"In-place operation not allowed: operands are distributed along different axes.
            Operand 1 with shape {t1.shape} is split along axis {t1.split}.
            Operand 2 with shape {t2.shape} is split along axis {t2.split}."
        )
```

How about this? The problem isn't that local shapes aren't broadcastable. They might be broadcastable, but still the operation would return the wrong result because it wouldn't be performed on the correct elements of the second operand.

Reply:
I have done the suggested modifications also at the other places where this error message appeared.